lib/CL/devices/pthread/pthread.c

```diff
@@ -117,6 +113,10 @@ struct data {
   mem_regions_management* mem_regions;
 #endif
 
+  /* Lock for command list related operations */
+  pthread_mutex_t cq_lock __attribute__ ((aligned (CACHELINE_SIZE)));
```

Comment:
> makes the false sharing problem worse.

Yes that's exactly what i meant to say. :)

 > If you don't need to share int i or you guard it with a mutex (the lock) then it should not be marked volatile

Yes. For some reason i haven't found out, it seems there's _a lot_ of variables in structs marked volatile, even ones not guarded with a mutex. I think this _might_ have something to do with the concept of inter-device handling of events. TBH this would explain volatile in structs like cl_event, but it doesn't explain why so many pthread driver internals are volatile.